openassessment/xblock/test/test_notifications.py

```diff
@@ -64,3 +71,122 @@ def test_send_staff_notification_error_logging(self, mock_send_event, mock_logge
 
         # Assertions
         mock_logger_error.assert_called_once_with(f"Error while sending ora staff notification: {mock_exception}")
+
+
+class TestSendGradeAssignedNotification(unittest.TestCase):
```

Comment:
There is a lot of mocking in these tests I am not sure if it covers issues with real data, will it be possible to test this with real data? I guess that implementation can be found in already existing tests.

Reply:
Yes, the tests for all the existing data already exist. 
In these tests, the sole purpose of all this data is to ensure that the notification event is being sent with the correct information, which is being achieved. 
We have no choice but to mock the relevant data as it is all being fetched through external functions, which are too difficult to mock and have already been tested separately.